extensions/ql-vscode/src/extension.ts

```diff
@@ -467,9 +468,11 @@ async function activateWithInstalledDistribution(
     selectedQuery: Uri | undefined,
     progress: ProgressCallback,
     token: CancellationToken,
+    databaseItem: DatabaseItem | undefined,
   ): Promise<void> {
     if (qs !== undefined) {
-      const dbItem = await databaseUI.getDatabaseItem(progress, token);
+      // If no databaseItem is specified, use the database currently selected in the Databases UI
+      const dbItem = databaseItem || await databaseUI.getDatabaseItem(progress, token);
```

Comment:
Minor: you could use only one variable throughout:
```ts
databaseItem = databaseItem || await databaseUI.getDatabaseItem(progress, token);
```
and replace `dbItem` in the two uses below with `databaseItem`.

Why do this? To avoid accidentally introducing code later that uses `databaseItem` when we should be using `dbItem`. (This is a classic mistake I have made often when two variables have similar names but only one variable has the fully initialised value...)

Reply:
Changed! Thanks for the tip—I can definitely see myself getting confused by variables otherwise 🙃